pydantic/schema.py

```diff
@@ -965,7 +966,9 @@ def go(type_: Any) -> Type[Any]:
                 used_constraints.update({'min_items', 'max_items'})
                 return conlist(go(args[0]), min_items=field_info.min_items, max_items=field_info.max_items)
 
-            if issubclass(origin, Set) and (field_info.min_items is not None or field_info.max_items is not None):
+            if issubclass(origin, AbstractSet) and (
```

Comment:
you seem to "remove" the difference between the treatment of `Set` and `FrozenSet` in this PR, have you seen this guy?
https://github.com/samuelcolvin/pydantic/blob/5261fd05a0374b84ce2602d45990baf480fa2417/pydantic/fields.py#L589
It looks like they are treated differently there and your PR does not touch it (idk if it is on purpose)

If I follow your logic, you'd want the behaviour to be consistent between `Set` and `Frozenset`. It is unclear to me why the validators would (optionally) apply to 1 and not in the other.

Forgive me if this does not make sense, I am new to pydantic and I do not yet know all the ins and outs.

Reply:
We are just talking about a quick fix for schema generation. The goal is not to go over all the validation and internals